text/0000-min-rust-version.md

```diff
@@ -0,0 +1,179 @@
+- Feature Name: min_rust_version
+- Start Date: 2018-06-28
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add `rust` field to the package section of `Cargo.toml` which will be used to
+specify crate's Minimum Supported Rust Version (MSRV):
+```toml
+[package]
+name = "foo"
+version = "0.1.0"
+rust = "1.30"
+```
+
+# Motivation
+[motivation]: #motivation
+
+Currently crates have no way to formally specify MSRV. As a result users can't
+check if crate can be built on their toolchain without building it. It also
+leads to the debate on how to handle crate version change on bumping MSRV,
+conservative approach is to consider such changes as breaking ones, which can
+hinder adoption of new features across ecosystem or result in version number
+inflation, which makes it harder to keep downstream crates up-to-date. More
+relaxed approach on another hand can result in broken crates for user of older
+compiler versions.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+`cargo init` will automatically create `Cargo.toml` with `rust` field equal to
+`rust="stable"` or `rust="nightly"` depending on the currently used toolcahin.
+On `cargo publish` cargo will take  currently used Rust compiler version and
+will insert it before uploading the crate. In other words localy your `Cargo.toml`
+willl still have `rust="stable"`, but version sent to crates.io will have
+`rust="1.30"` if you've used Rust 1.30. This version will be used to determine if
+crate can be used with the crate user's toolchain and to select appropriate
+dependency versions. In case if you have `rust="stable"`, but execute
+`cargo publish` with Nigthly toolcahin you will get an error.
+
+If you are sure that your crate supports older Rust versions (e.g. by using CI
+testing) you can change `rust` field accordingly. On `cargo publish` it will be
+checked that crate indeed can be built with the specified version. (though this
+check can be disabled with `--no-verify` option)
```

Comment:
What is the intended meaning of "checked that crate indeed can be built with the specified version", use `rustup` to install the specified version and package with it?

It seems like this can be covered by the error listed above:

> In case if you have `rust="stable"`, but execute `cargo publish` with Nigthly toolcahin you will get an error.

if you have `rust = "1.30"` then you will only be allowed to publish when using Rust v1.30.

Reply:
I had in mind that `cargo publish` in addition to the current toolchain will build crate with MSRV toolchain with some additional checks.But you are right, if we extend behaviour described for "stable"/"nightly" (i.e. you can `cargo publish only using MSRV toolchain), then describe checks will be redundant. I'll update RFC accordingly.